SPID/src/Distribute.cpp

```diff
@@ -12,128 +13,192 @@ namespace Distribute
-					} else {
-						auto idxIt = packageList.begin();
-						for (idxIt; idxIt != packageList.end(); ++idxIt) {
-							auto idx = std::distance(packageList.begin(), idxIt);
-							if (packageIdx == idx) {
-								break;
+						auto& packageList = npc->aiPackages.packages;
+						if (std::ranges::find(packageList, package) == packageList.end()) {
+							if (packageList.empty() || packageIdx == 0) {
+								packageList.push_front(package);
+							} else {
+								auto idxIt = packageList.begin();
+								for (idxIt; idxIt != packageList.end(); ++idxIt) {
+									auto idx = std::distance(packageList.begin(), idxIt);
+									if (packageIdx == idx) {
+										break;
+									}
+								}
+								if (idxIt != packageList.end()) {
+									packageList.insert_after(idxIt, package);
+								}
 							}
+							return true;
 						}
-						if (idxIt != packageList.end()) {
-							packageList.insert_after(idxIt, package);
+					} else if (a_packageOrList->Is(RE::FormType::FormList)) {
+						auto packageList = a_packageOrList->As<RE::BGSListForm>();
+
+						switch (packageIdx) {
+						case 0:
+							npc->defaultPackList = packageList;
+							break;
+						case 1:
+							npc->spectatorOverRidePackList = packageList;
+							break;
+						case 2:
+							npc->observeCorpseOverRidePackList = packageList;
+							break;
+						case 3:
+							npc->guardWarnOverRidePackList = packageList;
+							break;
+						case 4:
+							npc->enterCombatOverRidePackList = packageList;
+							break;
+						default:
+							break;
 						}
+						return true;
 					}
-					return true;
-				}
-			} else if (a_packageOrList->Is(RE::FormType::FormList)) {
-				auto packageList = a_packageOrList->As<RE::BGSListForm>();
-
-				switch (packageIdx) {
-				case 0:
-					npc->defaultPackList = packageList;
-					break;
-				case 1:
-					npc->spectatorOverRidePackList = packageList;
-					break;
-				case 2:
-					npc->observeCorpseOverRidePackList = packageList;
-					break;
-				case 3:
-					npc->guardWarnOverRidePackList = packageList;
-					break;
-				case 4:
-					npc->enterCombatOverRidePackList = packageList;
-					break;
-				default:
-					break;
-				}
-
-				return true;
-			}
-
-			return false;
-		});
 
-		for_each_form<RE::BGSOutfit>(a_npcData, Forms::outfits, a_input, [&](auto* a_outfit) {
-			if (npc->defaultOutfit != a_outfit && !npc->HasKeyword(processedOutfit)) {
-				npc->AddKeyword(processedOutfit);
-				npc->defaultOutfit = a_outfit;
-				return true;
-			}
-			return false;
-		});
+					return false;
+				},
+				accumulatedForms);
 
-		for_each_form<RE::BGSOutfit>(a_npcData, Forms::sleepOutfits, a_input, [&](auto* a_outfit) {
-			if (npc->sleepOutfit != a_outfit) {
-				npc->sleepOutfit = a_outfit;
-				return true;
-			}
-			return false;
-		});
+			for_each_form<RE::BGSOutfit>(
+				npcData, forms.outfits, input, [&](auto* a_outfit) {
+					if (npc->defaultOutfit != a_outfit && !npc->HasKeyword(processedOutfit)) {
```

Comment:
This check would cause linked outfits to fail (e.g. if `OutfitA` is linked to `OutfitB` when the latter will be distributed, it will block possibly desired overwrite by `OutfitA`)

Reply:
Fixed by introducing a new parameter `allowOverwrites` that is set to `true` for linked distribution. 🙂